src/index.d.ts

```diff
@@ -191,34 +191,17 @@ declare namespace preact {
 	 * Global options for preact
 	 */
 	interface Options {
-		/** Attach a hook that is invoked before render, mainly to check the arguments. */
-		root?(vnode: ComponentChild, parent: Element | Document | ShadowRoot | DocumentFragment): void;
```

Comment:
I think in user-land, most things you can do with `root` could be accomplished by just wrapping `render`. I think this pattern of composing functions (e.g. `let render = wrapper(render)`) is a much more powerful pattern that can do what this hook does and more, and so I think I'd encourage our users to do that over use this `root` hook.

Reply:
Agree, it's mainly needed for devtools and I doubt there is much use beyond that :+1: